pkg/config/config.go

```diff
@@ -672,15 +677,33 @@ func (c *baseConfig) resolve(builder pathBuilder, fsys fs.FS) error {
 			// Append supabase/ because paths in configs are specified relative to config.toml
 			function.Entrypoint = filepath.Join(builder.SupabaseDirPath, function.Entrypoint)
 		}
-		if len(function.ImportMap) == 0 {
+		if len(function.ImportMap) == 0 || strings.Contains(function.ImportMap, "deno.json") {
```

Comment:
```suggestion
		if len(function.ImportMap) == 0 {
```

Can we avoid this check for simplicity?

If users specify `import_map` in config.toml, then deno.json is treated simply as an import map, where no further references are resolved.

If user leaves `import_map` empty, and specifies `importMap` in deno.json without an overriding `imports` block, we can resolve the path reference to load an import map.

If user specifies both, then `import_map` in config.toml takes precedence.

Reply:
Also technically `importMap` can reference any file, not just the default location. If user references `../../import_map.json`, we need to bind mount both deno.json and `supabase/import_map.json`.

Reply:
I made a separate PR to implement this https://github.com/supabase/cli/pull/3430/commits/bae1d690980b0e295cf5f2cec4ff463592a16181 and also included some refactoring.

If you are happy with it, we can close this one.

Reply:
Sure! Looks good to me 🙏
Let's keep yours💚